askama_derive/src/lib.rs

```diff
@@ -62,8 +70,14 @@ pub(crate) fn build_template(ast: &syn::DeriveInput) -> Result<String, CompileEr
         eprintln!("{:?}", templates[input.path.as_path()].nodes());
     }
 
-    let code = Generator::new(&input, &contexts, heritage.as_ref(), MapChain::default())
-        .build(&contexts[input.path.as_path()])?;
+    let code = Generator::new(
+        &input,
+        &contexts,
+        heritage.as_ref(),
+        MapChain::default(),
+        input.block.is_some(),
```

Comment:
Given that we're already passing `input` to `Generator::new()`, there's no reason to pass `input.block.is_some()` separately -- it should derive the value for `WritableBuffer::discard`.